File: pay/src/main/java/net/yan100/compose/pay/service/PayService.java
package net.yan100.compose.pay.service;

import net.yan100.compose.pay.api.model.request.CreateOrderApiRequestParam;
import net.yan100.compose.pay.api.model.request.QueryOrderApiRequestParam;
import net.yan100.compose.pay.api.model.response.CreateOrderApiResponseResult;
import net.yan100.compose.pay.api.model.response.QueryOrderApiResponseResult;

public interface PayService {

  CreateOrderApiResponseResult createOrder(CreateOrderApiRequestParam createOrderRequestParam);

  QueryOrderApiResponseResult queryOrder(QueryOrderApiRequestParam queryOrderApiRequestParam);

}

File: pay/src/main/kotlin/net/yan100/compose/pay/service/PayService.kt
package net.yan100.compose.pay.service

import net.yan100.compose.pay.models.request.CreateOrderApiRequestParam
import net.yan100.compose.pay.models.request.QueryOrderApiRequestParam
import net.yan100.compose.pay.models.response.CreateOrderApiResponseResult
import net.yan100.compose.pay.models.response.QueryOrderApiResponseResult

interface PayService {
  fun createOrder(createOrderRequestParam: CreateOrderApiRequestParam): CreateOrderApiResponseResult?
  fun queryOrder(queryOrderApiRequestParam: QueryOrderApiRequestParam): QueryOrderApiResponseResult?
}
